### ISSUE TITLE
Couldn't find the absolute location ${toolsjarSystemPath}.

### ISSUE THREAD

**Issue by mohanaraosv** [mojohaus/aspectj-maven-plugin#18]
I got the following error when i was trying to import a new project. I couldn't find the reason why it been referred from jre instead of jdk.

Problem : 1 problem was encountered while building the effective model for org.codehaus.mojo:aspectj-maven-plugin:1.8 [ERROR] 'dependencies.dependency.systemPath' for com.sun:tools:jar must specify an absolute path but is ${toolsjarSystemPath} 

ref : http://stackoverflow.com/questions/33208876/maven-error-after-importing-latest-code-from-github
 
**Name :java.home  Value : C:\Program Files\Java\jre1.8.0_40**

Specified profiles in aspectj-maven-plugin.
```
<profile>
      <id>standardToolsJar-profile</id>
      <activation>
        <activeByDefault>true</activeByDefault>
        <file>
          <exists>${java.home}/../lib/tools.jar</exists>
        </file>
      </activation>
      <properties>
        <toolsjarSystemPath>${java.home}/../lib/tools.jar</toolsjarSystemPath>
      </properties>
    </profile>
    <profile>
      <id>appleJdkToolsJar-profile</id>
      <activation>
        <activeByDefault>false</activeByDefault>
        <file>
          <exists>${java.home}/../Classes/classes.jar</exists>
        </file>
      </activation>
      <properties>
        <toolsjarSystemPath>${java.home}/../Classes/classes.jar</toolsjarSystemPath>
      </properties>
    </profile>
    <profile>
      <id>java8</id>
      <activation>
        <jdk>1.8</jdk>
      </activation>
      <properties>
        <additionalparam>-Xdoclint:none</additionalparam>
      </properties>
    </profile>
```

**Comment by kriegaex** [comment]
I am also affected by this problem and think I just fixed it in my pull request. Would you consider releasing 1.9 soon or maybe cherry-pick my small commit onto 1.8 and push a bugfix release?

**Comment by davidkarlsen** [comment]
@mohanaraosv  I just deployed 1.9-SNAPSHOT.
Please give it a spin and if it works out I'll release.

**Comment by mohanaraosv** [comment]
I'm still facing the same problem.  :( 

1 problem was encountered while building the effective model for org.codehaus.mojo:aspectj-maven-plugin:**1.9-SNAPSHOT** [ERROR] 'dependencies.dependency.systemPath' for com.sun:tools:jar must specify an 
 absolute path but is ${toolsjarSystemPath} @

**Comment by kriegaex** [comment]
Sorry @davidkarlsen, I am on the road and cannot test your snapshot before the end of next week. I tested it with mine though, the one I created the PR from. That one worked.

@mohanaraosv: Are you using the right snapshot from the right repository and did you refresh the snapshot? Maybe you want to delete it from your local Maven repo and try again. Just guessing...

**Comment by kriegaex** [comment]
I just re-tested, sorry for the long delay. I have no idea what changed since I last tested my own modification because then it worked, now it does not work anymore, no matter whether I build the plugin locally or use @davidkarlsen 's version from

    <pluginRepositories>
        <pluginRepository>
            <id>maven-snapshots</id>
            <name>Maven Snapshots</name>
            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
            <snapshots>
                <enabled>true</enabled>
            </snapshots>
        </pluginRepository>
    </pluginRepositories>

At the moment I am clueless, I have to investigate the issue further. Any ideas would be welcome while I am trying to remember what I did four weeks ago that seemed to make it work.

**Comment by kriegaex** [comment]
Okay, this time I guess I really fixed it. The problem was different from what I initially thought and last time it only worked for me because I had some special variables set locally. Sorry for that. Now I have reverted my previous commit because it was not really necessary. I replaced it with another one providing a well-explained workaround (see _pom.xml_ from my PR).

What broke the plugin for Eclipse since version 1.8 was the new _java8_ Maven profile because it defines a property. This somehow shadows (thus hides) the former _toolsjarSystemPath_ property defined in one of the other profiles instead of logically merging them before importing/refreshing AspectJ Maven projects in Eclipse. This is really an Eclipse bug, but my workaround fixes it until it might be fixed upstream. Maybe you want to file a ticket for Eclipse or AJDT. I do not really know the source of the problem.
